File: done/bts16p3.py
import sys
input = sys.stdin.readline

n = int(input())
last = ""
last_value = 1
ans = 0
for i, name in enumerate(input().split()):
    if name[0] == last:
        last_value += 1
        ans += last_value
    else:
        ans += 1
        last_value = 1
    last = name[0]
print(ans)

File: done/bts16p3.c
#include <stdio.h>
#include <stdlib.h>

int n, last_value = 1, ans;
char temp[21], last = ' ';

int main() {
    scanf("%d", &n);
    while (n--) {
        scanf("%s ", temp);
        if (temp[0] == last) {
            last_value++;
            ans += last_value;
        } else {
            ans++;
            last_value = 1;
        }
        last = temp[0];
    }
    printf("%d\n", ans);
    return 0;
}